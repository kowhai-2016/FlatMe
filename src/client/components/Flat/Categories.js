import React from 'react'
import { Link } from 'react-router'

import { Navbar } from 'react-bootstrap'

const Categories = props => {
  const id = props.id
  return (
    <Navbar inverse className='categories-bar'>
      <div><img src='/images/phone.svg' width='25px' /><Link to={`/flat/${id}`}> Contacts</Link></div>
      <div><img src='/images/paper.svg' width='25px' /><Link to={`/flat/${id}/documents`}> Documents</Link></div>
      <div><img src='/images/interface-1.svg' width='25px' /><Link to={`/flat/${id}/notes`}> Notes</Link></div>
    </Navbar>
  )
}

export default Categories
