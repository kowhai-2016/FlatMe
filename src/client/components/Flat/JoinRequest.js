import React, { PropTypes } from 'react'
import { Link } from 'react-router'

const JoinRequest = props => {
  const request = props.request
  const user = request.user
  const { id: userId, firstName, lastName } = user
  const accept = () => props.acceptJoinRequest(request.id)
  const ignore = () => props.ignoreJoinRequest(request.id)
  return (
    <li>
      <Link to={`/user/${userId}`}>
        {firstName} {lastName}
      </Link>
      <div>
        <button
          className='btn btn-success glyphicon glyphicon-ok'
          onClick={accept}
          >
          Accept
        </button>
        <button
          className='btn btn-danger glyphicon glyphicon-remove'
          onClick={ignore}
          >
          Ignore
        </button>
      </div>
    </li>
  )
}

JoinRequest.propTypes = {
  acceptJoinRequest: PropTypes.func.isRequired,
  ignoreJoinRequest: PropTypes.func.isRequired,
  request: PropTypes.shape({
    id: PropTypes.number.isRequired,
    user: {
      id: PropTypes.number.isRequired,
      firstName: PropTypes.string.isRequired,
      lastName: PropTypes.string.isRequired
    }.isRequired
  }).isRequired
}

export default JoinRequest
