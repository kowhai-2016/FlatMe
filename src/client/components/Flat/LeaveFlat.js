import React from 'react'
import { Button, Modal } from 'react-bootstrap'

export default React.createClass({
  getInitialState () {
    return {show: false}
  },
  open () {
    this.setState({show: true})
  },
  close () {
    this.setState({show: false})
  },
  render () {
    return (
      <div className='LeaveFlat'>
        <div className='LeaveButton'>
          <Button onClick={this.open}>Leave this flat</Button>
        </div>
        <Modal className='CreateFlatModal' show={this.state.show} onHide={this.close}>
          <Modal.Header closeButton>
            <Modal.Title>Are you sure you want to leave this flat?</Modal.Title>
          </Modal.Header>
          <Modal.Footer>
            <Button className='Yes' onClick={this.props.leaveFlat}>Yes, I am sure</Button>
            <Button className='No' onClick={this.close}>Opps! No, don't remove me!</Button>
          </Modal.Footer>
        </Modal>
      </div>
    )
  }
})
