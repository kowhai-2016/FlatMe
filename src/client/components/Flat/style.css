/* Flat Dashboard page
–––––––––––––––––––––––––––––––––––––––––––––––––– */
.Flat .categories-bar {
  margin-top: 20px;
  padding-top: 20px;
  background-color: rgba(255,255,255,0.9);
  border: 2px solid lightseagreen;
}

.Flat .categories-bar div {
  padding-bottom: 10px;
  font-size: 1.2em;
}

.Flat .categories-bar div a {
  color: grey;
  text-align: center;
  text-decoration: none;
}

.Flat .categories-bar div a:hover {
  text-decoration: none;
  color: lightseagreen;
}

.Flat .categories-bar div .active {
  color: lightseagreen;
}

.Flat .popout {
  background-color: rgba(255,255,255,0.8);
  margin-left: 0;
  min-height: 900px;
  border-right-style: solid;
  border-left-style: solid;
  border-right-color: lightseagreen;
  border-right-width:medium;
  border-left-color: lightseagreen;
  border-left-width: medium;
}
