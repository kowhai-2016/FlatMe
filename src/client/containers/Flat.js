import { connect } from 'react-redux'
import { browserHistory } from 'react-router'

import { acceptJoinRequest, fetchFlat, ignoreJoinRequest, leaveFlat } from '../actions'
import Flat from '../components/Flat'

const mapStateToProps = (state, ownProps) => {
  const id = Number(ownProps.params.id)
  return {
    flat: state.flats[id],
    userId: state.account.user.id
  }
}

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    acceptJoinRequest (requestId) {
      dispatch(acceptJoinRequest(requestId))
        .then(() => {
          dispatch(this.fetchFlat())
        })
    },
    fetchFlat () {
      const id = Number(ownProps.params.id)
      dispatch(fetchFlat(id))
    },
    ignoreJoinRequest (requestId) {
      dispatch(ignoreJoinRequest(requestId))
        .then(() => {
          dispatch(this.fetchFlat())
        })
    },
    leaveFlat (userId, flatId) {
      dispatch(leaveFlat(userId, flatId))
        .then(() => {
          browserHistory.push('/')
        })
        .catch(error => {
          console.log(error)
        })
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Flat)
