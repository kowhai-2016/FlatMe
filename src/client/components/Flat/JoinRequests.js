import React, { PropTypes } from 'react'

import JoinRequest from './JoinRequest'

const JoinRequests = props => {
  return (
    <div>
      <h2>Join Requests</h2>
      <ul>
        {props.requests.map((request, index) => {
          return (
            <JoinRequest
              acceptJoinRequest={props.acceptJoinRequest}
              key={index}
              ignoreJoinRequest={props.ignoreJoinRequest}
              request={request}
              />
          )
        })}
      </ul>
    </div>
  )
}

JoinRequests.propTypes = {
  acceptJoinRequest: PropTypes.func.isRequired,
  ignoreJoinRequest: PropTypes.func.isRequired,
  requests: PropTypes.arrayOf(PropTypes.object).isRequired
}

export default JoinRequests
