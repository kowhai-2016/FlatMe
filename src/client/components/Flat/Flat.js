import React, { PropTypes } from 'react'
import { Row, Col } from 'react-bootstrap'

import Categories from './Categories'
import JoinRequests from './JoinRequests'
import LeaveFlat from './LeaveFlat'

import './style.css'

const Flat = React.createClass({
  componentDidMount () {
    this.props.fetchFlat()
  },

  leaveFlat () {
    this.props.leaveFlat(this.props.userId, this.props.flat.id)
  },

  render () {
    const flat = this.props.flat
    const flatName = flat ? flat.flatName : null
    const joinRequests = flat ? flat.requests : []
    const acceptJoinRequest = requestId => this.props.acceptJoinRequest(requestId)
    const ignoreJoinRequest = requestId => this.props.ignoreJoinRequest(requestId)
    return (
      <div className='Flat container-fluid plain-background flat'>
        <Row>
          <Col md={3}>
            <Categories id={this.props.params.id} />
          </Col>
          <Col md={7} className='popout'>
            <h1 className='text-center flat-name'>{flatName}</h1>
            {this.props.children && React.cloneElement(this.props.children, {
              flat
            }
          )}
          </Col>
          <Col md={3}>
            <LeaveFlat leaveFlat={this.leaveFlat} />
            <JoinRequests
              acceptJoinRequest={acceptJoinRequest}
              ignoreJoinRequest={ignoreJoinRequest}
              requests={joinRequests}
              />
          </Col>
        </Row>
      </div>
    )
  }
})

Flat.propTypes = {
  acceptJoinRequest: PropTypes.func.isRequired,
  fetchFlat: PropTypes.func.isRequired,
  flat: PropTypes.shape({
    flatName: PropTypes.string.isRequired
  }),
  ignoreJoinRequest: PropTypes.func.isRequired,
  params: PropTypes.shape({
    id: PropTypes.string.isRequired
  }).isRequired
}

export default Flat
