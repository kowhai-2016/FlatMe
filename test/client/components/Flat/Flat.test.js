/* eslint-env jest */

import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

import Flat from '../../../../src/client/components/Flat'

const mockStore = configureStore([])

test('Flat renders correctly', () => {
  const acceptJoinRequest = () => {}
  const fetchFlat = () => {}
  const ignoreJoinRequest = () => {}
  const params = {id: '1'}
  const store = mockStore({
    flats: {
      '1': {}
    }
  })
  const tree = renderer.create(
    <Provider store={store}>
      <Flat
        acceptJoinRequest={acceptJoinRequest}
        fetchFlat={fetchFlat}
        ignoreJoinRequest={ignoreJoinRequest}
        params={params}
        />
    </Provider>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
