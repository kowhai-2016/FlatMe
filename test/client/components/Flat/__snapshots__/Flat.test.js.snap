exports[`test Flat renders correctly 1`] = `
<div
  className="Flat container-fluid plain-background flat">
  <div
    className="row">
    <div
      className="col-md-3">
      <nav
        className="categories-bar navbar navbar-default"
        style={
          Object {
            "margin": "10%",
          }
        }>
        <div
          className="container">
          <div>
            <a
              onClick={[Function]}
              style={Object {}}>
              <img
                src="/images/phone.svg"
                width="25px" />
              <span>
                Contacts
              </span>
            </a>
          </div>
          <div>
            <a
              onClick={[Function]}
              style={Object {}}>
              <img
                src="/images/interface.svg"
                width="25px" />
              <span>
                Documents
              </span>
            </a>
          </div>
          <div>
            <a
              onClick={[Function]}
              style={Object {}}>
              <img
                src="/images/interface-1.svg"
                width="25px" />
              <span>
                Notes
              </span>
            </a>
          </div>
          <div>
            <a
              onClick={[Function]}
              style={Object {}}>
              <img
                src="/images/alarm.svg"
                width="25px" />
              <span>
                Calendar
              </span>
            </a>
          </div>
        </div>
      </nav>
    </div>
    <div
      className="popout col-md-7">
      <h1
        className="text-center" />
    </div>
    <div
      className="col-md-2">
      <div>
        <h2>
          Join Requests
        </h2>
        <ul />
      </div>
    </div>
  </div>
</div>
`;
